tests/gbs/test_similarity.py

```diff
@@ -194,3 +196,105 @@ def test_sample_max_count_deterministic(self, photon_num, count, monkeypatch):
+
+    orbits = {
+        ((1, 1, 2), 4): 12,
+        ((1, 1), 4): 6,
+        ((1, 2, 3), 4): 24,
+        ((1, 1, 1, 1), 5): 5,
+        ((1, 1, 2), 5): 30,
+        ((1, 2, 3), 5): 60,
+    }
+
+    calc_cardinalities = {}
+    for o, _ in orbits.items():
+        calc_cardinalities[o] = similarity.orbit_cardinality(*o)
+
+    assert calc_cardinalities == orbits
+
+
+def test_event_cardinality():
+    """Test if function ``strawberryfields.gbs.similarity.event_cardinality`` returns the
+    correct number of samples for some hard-coded examples."""
+
+    events = {
+        (5, 3, 6): 216,
+        (6, 3, 6): 336,
+        (5, 2, 6): 126,
+        (5, 3, 7): 413,
+        (6, 3, 7): 728,
+        (5, 2, 7): 266,
+    }
+
+    calc_cardinalities = {}
+    for o, _ in events.items():
+        calc_cardinalities[o] = similarity.event_cardinality(*o)
+
+    assert calc_cardinalities == events
+
+
+class TestProbOrbitMC:
+    """Tests for the function ``strawberryfields.gbs.similarity.p_orbit_mc.``"""
+
+    @pytest.mark.parametrize("n_mean", [0, 2, 4])
+    def test_actual_prob_orbit(self, n_mean):
+        """Tests if the output of the function is a valid probability between 0 and 1."""
+        graph = nx.complete_graph(10)
+
+        assert 0 <= similarity.p_orbit_mc(graph, [], n_mean) <= 1
+
+    def test_mean_computation_orbit(self, monkeypatch):
+        """Tests if the calculation of the sample mean is performed correctly. The test
+        monkeypatches the fock_prob function so that the probability is the same for each sample and
+         is equal to 1/5, i.e., one over the number of samples in the orbit [1,1,1,1] for 5 modes."""
+        graph = nx.complete_graph(5)
+        with monkeypatch.context() as m:
+            m.setattr("strawberryfields.backends.gaussianbackend.GaussianState.fock_prob",
+                      lambda *args, **kwargs: 0.2)
+            tol = 1e-5
+
+            assert np.abs(similarity.p_orbit_mc(graph, [1, 1, 1, 1]) - 1.0) < tol
+
+    def test_prob_vacuum_orbit(self):
+        """Tests if the function gives the right probability for the empty orbit when the GBS
+        device has been configured to have zero mean photon number."""
+        graph = nx.complete_graph(10)
+
+        assert similarity.p_orbit_mc(graph, [], 0) == 1.0
+
+
+class TestProbEventMC:
+
+    def test_prob_vacuum_event(self):
+        """Tests if the function gives the right probability for an event with zero photons when
+        the GBS device has been configured to have zero mean photon number."""
+        graph = nx.complete_graph(10)
+
+        assert similarity.p_event_mc(graph, 0, 0, 0) == 1.0
+
+    @pytest.mark.parametrize("n_mean", [0, 2, 4])
+    def test_actual_prob_event(self, n_mean):
+        """Tests if the output of the function is a valid probability between 0 and 1."""
+
+        graph = nx.complete_graph(10)
+
+        assert 0 <= similarity.p_event_mc(graph, 0, 0, 0) <= 1
```

Comment:
I'm also not sure about this probabilistic test, but if we have it do we need to replace the zeros in the above line?

Reply:
Removing probabilistic tests